package/utils/fritz-tools/src/fritz_cal_extract.c

```diff
@@ -154,53 +178,59 @@ struct cal_entry {
 int main(int argc, char **argv)
 {
 	struct cal_entry cal = { .len = 0 };
+	unsigned char *buf = NULL;
 	FILE *in = stdin;
 	FILE *out = stdout;
+	size_t datasize = DEFAULT_BUFFERSIZE;
 	size_t limit = 0, skip = 0;
 	int initial_offset = 0;
 	int entry = -1;
+	bool reversed = false, limit_was_set = true;
 	int ret;
 	int opt;
 
-	while ((opt = getopt(argc, argv, "s:e:o:l:i:")) != -1) {
+	while ((opt = getopt(argc, argv, "s:e:o:l:i:r")) != -1) {
 		switch (opt) {
-		case 's':
-			initial_offset = (int)get_num(optarg);
-			if (errno) {
-				perror("Failed to parse seek offset");
-				goto out_bad;
-			}
-			break;
-		case 'e':
-			entry = (int) htobe16(get_num(optarg));
-			if (errno) {
-				perror("Failed to entry id");
-				goto out_bad;
-			}
-			break;
-		case 'o':
-			out = fopen(optarg, "w");
-			if (!out) {
-				perror("Failed to create output file");
-				goto out_bad;
-			}
-			break;
-		case 'l':
-			limit = (size_t)get_num(optarg);
-			if (errno) {
-				perror("Failed to parse limit");
-				goto out_bad;
-			}
-			break;
-		case 'i':
-			skip = (size_t)get_num(optarg);
-			if (errno) {
-				perror("Failed to parse skip");
-				goto out_bad;
-			}
-			break;
-		default: /* '?' */
-			usage();
+			case 's':
+				initial_offset = (int)get_num(optarg);
+				if (errno) {
+					perror("Failed to parse seek offset");
+					goto out_bad;
+				}
+				break;
+			case 'e':
+				entry = (int) htobe16(get_num(optarg));
+				if (errno) {
+					perror("Failed to entry id");
+					goto out_bad;
+				}
+				break;
+			case 'o':
+				out = fopen(optarg, "w");
+				if (!out) {
+					perror("Failed to create output file");
+					goto out_bad;
+				}
+				break;
+			case 'l':
+				limit = (size_t)get_num(optarg);
+				if (errno) {
+					perror("Failed to parse limit");
+					goto out_bad;
+				}
+				break;
+			case 'i':
+				skip = (size_t)get_num(optarg);
+				if (errno) {
+					perror("Failed to parse skip");
+					goto out_bad;
+				}
+				break;
+			case 'r':
+				reversed = true;
+				break;
+			default: /* '?' */
+				usage();
```

Comment:
Don't reformat the whole switch statement, only add the new case.

Reply:
1. Changed commit to match indentation fix [#685e979d470a09361defe1612b001dfc3b7d9832](https://github.com/openwrt/openwrt/commit/685e979d470a09361defe1612b001dfc3b7d9832) from master branch.
2. Also deleted a piece of unreachable code:
```
index 550fd61cd1..c4547bc7e0 100644
--- a/package/utils/fritz-tools/src/fritz_cal_extract.c
+++ b/package/utils/fritz-tools/src/fritz_cal_extract.c
@@ -318,8 +318,6 @@ int main(int argc, char **argv)
 
 	goto out;
 
-	zerr(ret);
-
 out_bad:
 	ret = EXIT_FAILURE;
```